File: src/main/scala/com/vc/easy/L346.scala
package com.vc.easy

class MovingAverage(size: Int) {

  /** Initialize your data structure here. */
  import scala.collection.mutable
  val q = new mutable.Queue[Int]()

  def next(value: Int): Double = {
    if(q.size == size) {
      q.dequeue()
    }
    q.enqueue(value)
    q.sum.toDouble / q.size.toDouble
  }

}

/**
  * Your MovingAverage object will be instantiated and called as such:
  * var obj = new MovingAverage(size)
  * var param_1 = obj.next(`val`)
  */

File: src/main/java/com/vc/easy/L346.java
package com.vc.easy;

import java.util.*;

class MovingAverage {
    /** Initialize your data structure here. */
    List<Integer> list;
    double sum = 0;
    double size = 0;
    public MovingAverage(int size) {
        list = new LinkedList<>();
        this.size = size;
    }

    public double next(int val) {
        list.add(val);
        sum += val;
        if(list.size() <= size) return sum / list.size();
        else {
            sum -= list.remove(0);
            return sum / list.size();
        }
    }
}

/**
 * Your MovingAverage object will be instantiated and called as such:
 * MovingAverage obj = new MovingAverage(size);
 * double param_1 = obj.next(val);
 */
